x/authz/codec.go

```diff
@@ -22,3 +46,19 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 
 	msgservice.RegisterMsgServiceDesc(registry, MsgServiceDesc())
 }
+
+var (
+	amino = codec.NewLegacyAmino()
+
+	// ModuleCdc references the global x/authz module codec. Note, the codec should
+	// ONLY be used in certain instances of tests and for JSON encoding as Amino is
+	// still used for that purpose.
+	//
+	// The actual codec used for serialization should be provided to x/authz and
+	// defined at the application level.
+	ModuleCdc = codec.NewAminoCodec(amino)
```

Comment:
side note: this global codec will must bit us sooner or later.

Reply:
definitely better, although don't know how it will work with multi module setup... Amino codec will need to e it's own module.
Ideally this comes top-down as the proto codec. I gues we didn't do it because we thought amino will go away, and it's not worth to make that efforts. but amino doesn't go away.

Reply:
@robert-zaremba It is not possible to have a top-down approach in this case as the `sdk.Msg` interface has been poorly designed and it does not allow the `GetSignBytes` method to accept an external codec to be used. So it just assumes (and allows) each message to get its signature bytes differently, which doesn't make any sense honestly. 

The best approach would be to simply re-design how this entire process is handled, either requiring the `GetSignBytes` to accept an external codec or by finding any other way (maybe a custom Protobuf implementation that also deals with Amino correctly or something similar). 

In the meanwhile, we need to deal with what we have to make sure Ledger users are able to use the authz and feegrant modules properly.